Recast/PlayerViewController.swift

```diff
@@ -8,59 +8,344 @@
+        }
+
+        statusObserverToken = observe(\.player.currentItem?.status, options: [.new, .initial]) { (strongSelf, change) in
+            // Display an error if status becomes `.Failed`.
+            /*
+             Handle `NSNull` value for `NSKeyValueChangeNewKey`, i.e. when
+             `player.currentItem` is nil.
+             */
+            let newStatus: AVPlayerItem.Status
+
+            if let status = change.newValue {
+                newStatus = status!
+            } else {
+                newStatus = .unknown
+            }
+
+            if newStatus == .failed {
+                strongSelf.handleError(with: strongSelf.player.currentItem?.error?.localizedDescription, error: strongSelf.player.currentItem?.error)
+            }
+        }
+
+        currentObserverToken = observe(\.player.currentItem?, options: [.new, .initial]) { (strongSelf, _) in
+            strongSelf.updateNowPlayingInfo()
+        }
+
+        playerView.playerLayer.player = player
+
+        // Make sure we don't have a strong reference cycle by only capturing self as weak.
+        let interval = CMTimeMake(value: 1, timescale: 1)
+        timeObserverToken = player.addPeriodicTimeObserver(forInterval: interval, queue: DispatchQueue.main) { [weak self] time in
+            let timeElapsed = Float(CMTimeGetSeconds(time))
+
+            if let strongSelf = self, !strongSelf.controlsView.timeSlider.isSelected {
+                strongSelf.controlsView.timeSlider.value = Float(timeElapsed)
+                strongSelf.controlsView.leftTimeLabel.text = strongSelf.createTimeString(time: timeElapsed)
+            }
+        }
     }
 
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
-        // Dispose of any resources that can be recreated.
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+
+        if let timeObserverToken = timeObserverToken {
+            player.removeTimeObserver(timeObserverToken)
+            self.timeObserverToken = nil
+        }
     }
-}
 
-// MARK: - PlayerHeaderViewDelegate
-extension PlayerViewController: PlayerHeaderViewDelegate {
+    // MARK: - Player Controls
+
+    @objc func playPauseButtonWasPressed(_ sender: UIButton) {
+        if player.rate == 0.0 {
+            if currentTime == duration {
+                currentTime = 0.0
+            }
+            player.play()
+        } else {
+            player.pause()
+        }
+    }
+
+    private func skip(seconds: Double) {
+        guard let item = player.currentItem else { return }
+        let skipAmount = CMTime(seconds: seconds, preferredTimescale: CMTimeScale(1.0))
+        let newTime = CMTimeAdd(item.currentTime(), skipAmount)
+        player.currentItem?.seek(to: newTime, completionHandler: { success in
+            self.updateNowPlayingInfo()
+        })
+    }
+
+    @objc func skipBackButtonWasPressed(_ sender: UIButton) {
+        skip(seconds: -30)
```

Comment:
can we change 30 to a constant so that both `skipBack` and `skipForward` use the same value (negated in Back)